File: Backend/Data/ViewModels/Response/RouteVM.cs
using Backend.Data.ViewModels.Types;

namespace Backend.Data.ViewModels.Response
{
    /// <summary>
    /// View model class with the route response
    /// </summary>
    public class RouteVM
    {
        public DateTime CreateDate { get; set; }
        public List<CommentType> Comments { get; set; }
        public RouteResumeType RouteResume { get; set; }


        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="comments">Route's commnets</param>
        /// <param name="routeResume">Route's resume</param>
        public RouteVM(List<CommentType> comments, RouteResumeType routeResume, DateTime createDate)
        {
            this.Comments = comments;
            this.RouteResume = routeResume;
            this.CreateDate = createDate;
        }
    }
}

File: Frontend/src/app/viewModels/responses/RouteVM.ts
import { CommentType } from "../Types/CommentType";
import { RouteResumeType } from "../Types/RouteResumeType";

/**
 * View Model that represents a response when a route is requested
 */
export class RouteVM {

    /**
     * Construct's an instance of {@link RouteVM}
     * 
     * @param createDate route's creation date
     * @param routeResume route's resume
     * @param comments route's comment list
     */
    constructor(
        public createDate: Date,
        public routeResume: RouteResumeType,
        public comments?: Array<CommentType>
    ) { }
}